src/connector/src/sink.rs

```diff
@@ -0,0 +1,295 @@
+        // Perhaps better to replace this functionality with a new Sink trait method.
+        let conditions = |values: Vec<MySQLValue>| {
+            schema
+                .names()
+                .iter()
+                .zip_eq(values.iter())
+                .map(|(c, v)| format!("{}={}", c, v))
+                .collect::<Vec<String>>()
+        };
+
+        // Build a connection and start transaction
+        let endpoint = self.endpoint();
+        let mut endpoint = endpoint.split(':');
+        let mut builder = OptsBuilder::default()
+            .user(self.user())
+            .pass(self.password())
+            .ip_or_hostname(endpoint.next().unwrap())
+            .db_name(self.database());
+        // TODO(nanderstabel): Fix ParseIntError
+        if let Some(port) = endpoint.next() {
+            builder = builder.tcp_port(port.parse().unwrap());
+        }
+
+        let mut conn = Conn::new(builder).await?;
+        let mut transaction = conn.start_transaction(TxOpts::default()).await?;
+
+        let mut iter = chunk.ops().iter().enumerate();
+        while let Some((idx, op)) = iter.next() {
+            // Get SQL statement
+            let stmt = match *op {
+                Insert => format!(
+                    "INSERT INTO {} VALUES ({});",
+                    self.table(),
+                    join(values(idx)?, ",")
+                ),
+                Delete => format!(
+                    "DELETE FROM {} WHERE ({});",
+                    self.table(),
+                    join(conditions(values(idx)?), " AND ")
+                ),
+                UpdateDelete => {
+                    if let Some((idx2, UpdateInsert)) = iter.next() {
+                        format!(
+                            "UPDATE {} SET {} WHERE {};",
+                            self.table,
+                            join(conditions(values(idx2)?), ","),
+                            join(conditions(values(idx)?), " AND ")
+                        )
+                    } else {
+                        panic!("UpdateDelete should always be followed by an UpdateInsert!")
+                    }
+                }
+                _ => panic!("UpdateInsert should always follow an UpdateDelete!"),
+            };
+            transaction.exec_drop(stmt, Params::Empty).await?;
+        }
+
+        // Commit and drop the connection.
+        transaction.commit().await?;
+        drop(conn);
+        Ok(())
+    }
+
+    fn endpoint(&self) -> String {
+        self.endpoint.clone()
+    }
+
+    fn table(&self) -> String {
+        self.table.clone()
+    }
+
+    fn database(&self) -> Option<String> {
+        self.database.clone()
+    }
+
+    fn user(&self) -> Option<String> {
+        self.user.clone()
+    }
+
+    fn password(&self) -> Option<String> {
+        self.password.clone()
+    }
+}
+
+pub struct RedisSink;
+
+#[async_trait]
+impl Sink for RedisSink {
```

Comment:
/sink/mysql.rs
/sink/redis.rs
/sink/mod.rs

Personally recommend this way to organize the files.